support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/ticket/refreshtoken/OAuth20DefaultRefreshTokenFactory.java

```diff
@@ -77,7 +82,9 @@ private ExpirationPolicy determineExpirationPolicyForService(final String client
             val policy = registeredService.getRefreshTokenExpirationPolicy();
             val timeToKill = policy.getTimeToKill();
             if (StringUtils.isNotBlank(timeToKill)) {
-                return new OAuth20RefreshTokenExpirationPolicy(Beans.newDuration(timeToKill).getSeconds());
+                val timeToKillInSeconds = Beans.newDuration(timeToKill).getSeconds();
+                return shouldTGTExpirationTriggerRTExpiration ? new OAuth20RefreshTokenExpirationPolicy(timeToKillInSeconds) :
```

Comment:
There are also a few other considerations here:

1. The code that creates the policy here for the service seems to be duplicating what `expirationPolicyBuilder.buildTicketExpirationPolicy()` does. We should try to avoid this, if possible.
2. I think this `shouldTGTExpirationTriggerRTExpiration` should be a property of the policy that one can define and override per service, and if left undefined, it could then fall back to the default server behavior. What you have now here does allow me to override the tracking mechanism per service as well, which is something you may want to allow a single RP.

Reply:
OK. I will revisit this part to avoid duplication...

Reply:
When you say: "I think this shouldTGTExpirationTriggerRTExpiration should be a property of the policy", what policy do you mean?

Reply:
I had in mind that it would be something like this:

```json
{
  "@class" : "org.apereo.cas.support.oauth.services.OAuthRegisteredService",
  "clientId": "clientid",
  "clientSecret": "clientSecret",
  "serviceId" : "^(https|imaps)://<redirect-uri>.*",
  "name" : "OAuthService",
  "id" : 100,
  "refreshTokenExpirationPolicy": {
    "@class": "org.apereo.cas.support.oauth.services.DefaultRegisteredServiceOAuthRefreshTokenExpirationPolicy",
    "timeToKill": "PT100S",
    "track": true // better worded obviously; track is defined as Boolean (uppercase B) or TristateBoolean
  }
}
```

On second thought, if we do this that would mean that we'd have to do the same thing for ATs, and pretty much any other child ticket whose expiration policy can be controlled per service. Too much? Necessary?

Reply:
OK. I see. I understand your comment, but I think this is something different. Currently, we have a global tracking and a global and per service expiration policy.

You propose to add tracking per service here. This would be a new feature.

This is not what this PR is about. Here, I only intend to fix a bug when someone defines the RT expiration policy at the service level (it works at the global level).

Does it make sense?

Reply:
That is totally fair, sure. Please proceed

Reply:
OK. I will keep this part as is.

Struggling in my Puppeteer tests merge, I just realized that the `trackDescendantTickets` is `true` by default: I think we should keep it `false` by default as it used to be, shouldn't we?

Reply:
By default this was true, I think. Tickets were always tracked and there was no option to control the tracking. The removal setting was off by default perhaps.

Reply:
Oh I see.

Yes, tickets were tracked but it was useless as the removal setting was off by default.

As they are now coupled (and the track setting replaces the removal setting), the track setting should be off by default to keep the same logout behaviour and save space in the tickets.

I will go this way, refactor the Puppeteer tests and fix the OAuth tests.

Stay tuned.